internal/controller/nonadminbackup_controller.go

```diff
@@ -128,125 +121,132 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
-	logger := logrLogger.WithValues("ValidateVeleroBackupSpec", nab.Namespace)
+// The function validates the Spec from the NonAdminBackup object.
+// If the BackupSpec is invalid, the function sets the NonAdminBackup phase to "BackingOff".
+// If the BackupSpec is invalid, the function sets the NonAdminBackup condition Accepted to "False".
+// If the BackupSpec is valid, the function sets the NonAdminBackup condition Accepted to "True".
+func (r *NonAdminBackupReconciler) ValidateSpec(ctx context.Context, logrLogger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (exitReconcile bool, requeueReconcile bool, errorReconcile error) {
+	logger := logrLogger
 
 	// Main Validation point for the VeleroBackup included in NonAdminBackup spec
 	_, err := function.GetBackupSpecFromNonAdminBackup(nab)
-
 	if err != nil {
-		// Use errMsg if errMsgFromErr is not available, otherwise use errMsgFromErr
-		errMsg := "NonAdminBackup CR does not contain valid BackupSpec"
-		if errMsgFromErr := err.Error(); errMsgFromErr != "" {
-			errMsg = errMsgFromErr
-		}
-		logger.Error(err, errMsg)
-
-		updatedStatus, errUpdateStatus := function.UpdateNonAdminPhase(ctx, r.Client, logger, nab, nacv1alpha1.NonAdminBackupPhaseBackingOff)
-		if errUpdateStatus != nil {
-			logger.Error(errUpdateStatus, "Unable to set NonAdminBackup Phase: BackingOff", nameField, nab.Name, constant.NameSpaceString, nab.Namespace)
-			return true, false, errUpdateStatus
-		} else if updatedStatus {
-			// We do not requeue - the State was set to BackingOff
-			return true, false, nil
+		updated := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseBackingOff)
+		if updated {
+			if updateErr := r.Status().Update(ctx, nab); updateErr != nil {
+				logger.Error(updateErr, phaseUpdateError)
+				return true, false, updateErr
+			}
+
+			logger.V(1).Info(phaseUpdateRequeue)
+			return false, true, nil
 		}
 
-		// Continue. VeleroBackup looks fine, setting Accepted condition
-		updatedCondition, errUpdateCondition := function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionFalse, "InvalidBackupSpec", errMsg)
-
-		if errUpdateCondition != nil {
-			logger.Error(errUpdateCondition, "Unable to set BackupAccepted Condition: False", nameField, nab.Name, constant.NameSpaceString, nab.Namespace)
-			return true, false, errUpdateCondition
-		} else if updatedCondition {
-			return true, false, nil
+		updated = meta.SetStatusCondition(&nab.Status.Conditions,
+			metav1.Condition{
+				Type:    string(nacv1alpha1.NonAdminConditionAccepted),
+				Status:  metav1.ConditionFalse,
+				Reason:  "InvalidBackupSpec",
+				Message: err.Error(),
+			},
+		)
+		if updated {
+			if updateErr := r.Status().Update(ctx, nab); updateErr != nil {
+				logger.Error(updateErr, conditionUpdateError)
```

Comment:
What do you mean by update log? There was an error in updating the object, so there should not be separate call to the k8s api as this could lead to another error, we should either retry or exit.